tkg/managementcomponents/management_component_install.go

```diff
@@ -290,14 +290,18 @@ func InstallManagementComponents(mcip *ManagementComponentsInstallOptions) error
 
 // InstallKappController installs kapp-controller to the cluster
 func InstallKappController(clusterClient clusterclient.Client, kappControllerOptions KappControllerOptions) error {
+	// Ensure last-applied annotation is present before kapp-controller upgrade
+	if err := clusterClient.WaitForDeployment(constants.KappControllerDeploymentName, kappControllerOptions.KappControllerInstallNamespace); err == nil { // only attempt to add annotation when kapp-controller exists on the cluster
```

Comment:
Does this function only get called for management cluster ? Should we skip workload cluster here too ?

Reply:
Yes, this function and also function `InstallOrUpgradeKappController` ([here](https://github.com/vmware-tanzu/tanzu-framework/blob/main/tkg/client/management_components.go#L25)) is only called for management clusters.